### ISSUE TITLE
@:structInit vs arguments of basic types with default values

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#9177]
```haxe
@:structInit
class C {
	public function new(x:Int = 0) {}
}

class Main {
	static function main() {
		var c:C = {};
	}
}
```

Compiling this to a static target will fail with the following error (repeated twice for some reason):
```
Main.hx:8: characters 13-15 : On static platforms, null can't be used as basic type Int
Main.hx:8: characters 13-15 : On static platforms, null can't be used as basic type Int
```

**Comment by Simn** [comment]
I'm seeing this on the CI:

```
results: SOME TESTS FAILURES (success: false)
unit.issues.Issue9177
  test: FAILURE F
    line: 7, expected 123 but it is 0
```

**Comment by RealyUniqueName** [comment]
Fixed in 393732bc50f3111eb017eab9b868b0e61cdeab54
